infra/scripts/cdc-provisioning.sh

```
@@ -19,16 +75,26 @@ register_connector() {
-    RESPONSE=$(curl -s -o /dev/null -w "%{http_code}" -X POST -H "Content-Type: application/json" --data @"$FILE" http://kafka-connect:8083/connectors)
-    if [ "$RESPONSE" -eq 201 ]; then
-      echo "✅ Đăng ký $NAME thành công."
+    # Render connector JSON safely (supports special characters)
+    TMP_FILE="$(mktemp)"
+    render_connector_payload "$FILE" > "$TMP_FILE"
+
+    RESPONSE=$(curl -s -w "%{http_code}" --connect-timeout 5 --max-time 15 -X POST -H "Content-Type: application/json" --data @"$TMP_FILE" http://kafka-connect:8083/connectors)
+    rm -f "$TMP_FILE"
+    HTTP_CODE=$(printf '%s' "$RESPONSE" | tail -c 3)
+    BODY=${RESPONSE%???}
+    if [[ "$HTTP_CODE" =~ ^[0-9]+$ ]] && { [ "$HTTP_CODE" -eq 201 ] || [ "$HTTP_CODE" -eq 200 ]; }; then
```

Shellcheck (0.11.0)

[warning] In POSIX sh, [[ ]] is undefined.

(SC3010)